runtime/src/util/batch_return.rs

```diff
@@ -49,9 +49,9 @@ impl BatchReturn {
         ret
     }
 
-    // Returns a subset of items corresponding to the successful indices.
-    // Panics if `items` is not the same length as this batch return.
-    pub fn successes<T: Copy>(&self, items: &[T]) -> Vec<T> {
+    /// Returns a subset of items corresponding to the successful indices.
+    /// Panics if `items` is not the same length as this batch return.
+    pub fn successes<T: Clone>(&self, items: &[T]) -> Vec<T> {
```

Comment:
These clones might be more than we need. Can we return `Vec<&T>` and defer any cloning to the caller?

Low priority, feel free to skip this if it doesn't go smoothly.